lib/pbench/agent/results.py

```diff
@@ -226,24 +246,18 @@ def make_result_tb(self, single_threaded: bool = False) -> TarballRecord:
                     xz_proc.stdin.close()
                     xz_proc.wait()
         except Exception as exc:
-            msg = f"Tar ball creation failed for {self.result_dir}, skipping {exc}"
-            try:
-                tarball.unlink()
-            except FileNotFoundError:
-                pass
-            except Exception as unlink_exc:
-                msg = f"{msg}; also encountered an error while removing failed tar ball, {tarball}: '{unlink_exc}'"
+            msg = self._unlink_tarball(
+                tarball,
+                f"Tar ball creation failed for {self.result_dir}, skipping {exc}",
```

Comment:
This comes out to `skipping IOERROR` or whatever, which is not very clear. Again, I think `"Skipping {self.result_dir} because of error {exc}"` would be a lot more clear.